controllers/storagecluster/generate.go

```diff
@@ -79,7 +80,23 @@ func generateNameForSnapshotClassDriver(initData *ocsv1.StorageCluster, snapshot
 	return fmt.Sprintf("%s.%s.csi.ceph.com", initData.Namespace, snapshotType)
 }
 
-func generateNameForSnapshotClassSecret(snapshotType SnapshotterType) string {
+func generateNameForSnapshotClassSecret(instance *ocsv1.StorageCluster, snapshotType SnapshotterType) string {
+	if instance.Spec.ExternalStorage.Enable {
+		data, ok := externalOCSResources[instance.UID]
+		if !ok {
+			log.Error(fmt.Errorf("Unable to retrieve external resource from externalOCSResources"),
+				"unable to generate name for snapshot class secret for external mode")
+		}
+		// print the Secret name which contains the prefix as the rook-csi-rbd/cephfs-provisioner default secret name
+		// for example if the secret name is rook-csi-rbd-node-rookStorage-replicapool it will check the prefix with rook-csi-rbd-node if it matches it will return that name
```

Comment:
This statement is a little unclear: "... it will return **that** name..." 
Since there are two secret names being discussed here, "that" is unclear to me.